lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -1024,163 +1024,6 @@ bool BorrowedValue::visitInteriorPointerOperandHelper(
   return true;
 }
 
-// FIXME: This does not yet assume complete lifetimes. Therefore, it currently
-// recursively looks through scoped uses, such as load_borrow. We should
-// separate the logic for lifetime completion from the logic that can assume
-// complete lifetimes.
```

Comment:
Was this FIXME addressed?

Reply:
This FIXME is actually related to how this API is used by some older Ownership SSA APIs, not APIs that are just attempting to use it to visit addresses and their transitive users. My change refactors the visitor construct so that it can express both the expected behavior of those Ownership SSA APIs and also handle use cases where we just want a simple address walker (which the move checker needed). Since the actual change needed in question here was to expose this address walker functionality rather than fix issues in the Ownership SSA based APIs, it made sense to just ensure that I preserved the current behavior for those APIs.

I did this by provided 2 versions of findTransitiveUsesForAddress:

1. A new one that takes a visitor construct and just visits transitive addresses and their users.
2. Then I have a 2nd API that has the original function signature and that is called by the Ownership SSA APIs. The 2nd API calls the 1st API with a visitor struct that preserves the original behavior of the API (and what the Ownership SSA API users expect) by having its visitor struct when it visits a use that is a scoped instruction, append the scope ending instructions instead.

Reply:
Basically, the FIXME says that this code is over-complicated because we still haven't enabled complete lifetimes. There are many such places to be cleaned up later. it should just have been a TODO